### ISSUE TITLE
File List Selection via Up / Down Arrow Unreliable

### ISSUE THREAD

**Issue by PaltryProgrammer** [dnGrep/dnGrep#992]
greetings kind regards
attached please find GIF demonstrate slow / unreliable response upon File List Selection . please note the GIF will not seem to animate for some time to demonstrate slow response . please note the duration of the GIF is 2m 16s . 
thank you kindly
![File List Up-Down Arrow Keys Unreliable](https://github.com/dnGrep/dnGrep/assets/48535920/91443ee2-151a-4528-9111-b0f91c8bed45)

**Comment by doug24** [comment]
It looks like it is the length of time it takes to load that one file "f(8).txt" in the preview window - just how big is it?  The thumb on the preview window is pretty small, I'm guessing that line 10 starting with 'var' is all one long line comprising the remainer of the file.  Long lines load much more slowly than many shorter lines.

**Comment by PaltryProgrammer** [comment]
greetings kind regards
thank you for providing dnGrep . i utilize it often .
re/ above posted matter f(8) file is size 394548 . line 10 is size 374597 .
re/ file f(8) load duration may i suggest you examine your code as i have confirmed my own and several other editors load file f(8) in an instant w/ no discernable delay . in my own code i have experimented w/ various buffer sizes to wit 1Mi 1Ki and even a tiny 0x10 . each were as stated id est quick . a screen-shot of my test code and timing results of these tests are attached . in these tests proper correct loading of the file was confirmed . also attached are GIFs of three editors 010 Editor BowPad and NotePad++ each demonstrate loading file f(8) . as you can see they also are quick . 
![image](https://github.com/dnGrep/dnGrep/assets/48535920/6e489aa6-23f4-4e16-b11a-65caae5eab61)
![Fast Load of f(8) 010 Editor](https://github.com/dnGrep/dnGrep/assets/48535920/8978f216-9b97-4770-833b-5a92fa70dc38)
![Fast Load of f(8) BowPad](https://github.com/dnGrep/dnGrep/assets/48535920/9e6b07df-6fd2-4a55-813c-b33f3fffea3c)
![Fast Load of f(8) NotePad++](https://github.com/dnGrep/dnGrep/assets/48535920/7b66d1bc-ab80-49b4-a6bf-fa98f95ca2b0)

**Comment by doug24** [comment]
If you uncheck 'Preview file' I am pretty sure that you will find that moving up and down the results list will be instantaneous, and I think that dnGrep does a pretty good job of searching in large files. 

I believe the problem is in the preview control (which is a third-party control AvalonEdit).  I already have a fork of AvalonEdit so there _may_ be something I can do without waiting for the library authors to possibly make a change at some time.

Is there any reason you did not post the problem file here?  It would make my task much easier to have a file that I can use to reproduce the problem.

**Comment by PaltryProgrammer** [comment]
thank you for your kind assistance . attached is file f(8) as requested . kind regards .
[f(8).txt](https://github.com/dnGrep/dnGrep/files/13404584/f.8.txt)

**Comment by doug24** [comment]
Thank-you!  
Just another observation - having word wrap enabled in the preview adds significantly to the delay.

**Comment by doug24** [comment]
This is an old, [known issue](https://github.com/icsharpcode/AvalonEdit/issues/11) with AvalonEdit.  

> This is a known problem. AvalonEdit isn't really designed to handle long lines, and WPF has some strange performance bugs with them -- I tested this some time ago, and WPF rendering time increases linearly with the line length up to some strange font-dependent limit (this might be the 9600 in your case), where it will suddenly start to word-wrap. Increasing the line length above this limit then causes quadratic growth in the time taken.

**Comment by doug24** [comment]
Yes, this is a defect, but it's not something I can fix, and I doubt it will ever be fixed in the underlying controls in AvalonEdit or WPF. I don't think there will be any way to show long lines without changing to a non-WPF control.  

Maybe this isn't the ideal result, but I have added code to truncate the visual display of lines in the preview window to 8000 characters. With this, the control loads without delay and there is no delay in changing from one file to another.  I chose 8000 characters because it is under the threshold where the WPF control starts behaving strangely.

I did try to add a user option to disable truncating long lines, but that seemed to only make the rendering worse.

I'm OK with this approach because I think this is an edge case: a minified JavaScript file is not intended to be human readable text. Yes, other editors can open the file quickly without any problem - so we can take advantage of that and can use the external editor of your choice to view the full content. dnGrep can search the file quickly and tell you what lines and how many times it is found and at what offset on the line. Or just use the external editor's search function to find the individual occurrences in the long line.  

When the visual display is truncated, the file content is unchanged, only the display rendering is changed.  If you select a line and copy it, you will get the whole line on the clipboard.
